nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/ResultSetRecordSet.java

```diff
@@ -297,7 +271,54 @@ private DataType getDataType(final int sqlType, final ResultSet rs, final int co
         }
     }
 
-    private static DataType getArrayBaseType(final Array array) throws SQLException {
+    private DataType getArrayDataType(ResultSet rs, int columnIndex, boolean useLogicalTypes) throws SQLException {
+        // The JDBC API does not allow us to know what the base type of an array is through the metadata.
+        // As a result, we have to obtain the actual Array for this record. Once we have this, we can determine
+        // the base type. However, if the base type is, itself, an array, we will simply return a base type of
+        // String because otherwise, we need the ResultSet for the array itself, and many JDBC Drivers do not
+        // support calling Array.getResultSet() and will throw an Exception if that is not supported.
+        if (rs.isAfterLast()) {
```

Comment:
There are merge conflicts due to a try-catch added here and some other changes via [NIFI-8376](https://issues.apache.org/jira/browse/NIFI-8376)

Reply:
@mattyb149 I rebased and merged these modifications. Unfortunately I managed to do that by deleting this branch and opening a new one with the same name, so I had to create a new PR: https://github.com/apache/nifi/pull/5388
Please have a look at that if your time permits. Thank you!